Fix docker build

Ashrith-Shetty opened the pull request:
<!--- Provide a general, concise summary of your changes in the Title above -->

- Using python3.7 to run OWTF, since python3.8 had dependency issues. 
- Changed creation and activation of Python Virtual Environment.
- Added wait-for-it.sh which was missing in Docker folder.
- Updated configuration in settings.py to connect with PostgreSQL container.
- Changed Kali Linux image to kalilinux/kali.



## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
https://github.com/owtf/owtf/issues/1023

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
After the change, you can successfully build the docker container and run OWTF.

## Reviewers
<!--- @mentions of the person/people responsible for reviewing proposed changes. -->
@viyatb @sharmamohit123 

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style (modified PEP8) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Ashrith-Shetty committed:
Added wait-for-it.sh

wait-for-it.sh script was missing in the docker folder.

Ashrith-Shetty committed:
Fix docker build issues

- Using python3.7 to run OWTF, since python3.8 had dependency issues. 
- Added wait-for-it.sh which was missing in Dockerfile but present in docker-compose.dev.yml. 
- A simple and more elegant way for activating Python Virtual Environment.

Ashrith-Shetty committed:
Update settings.py

Add database configuration when used within a docker container.

Ashrith-Shetty committed:
Update base.txt

Updating the version of six to make it compatible with python3.7

viyatb commented on docker/Dockerfile:
```diff
@@ -23,38 +23,44 @@ RUN apt-get -y install  xvfb xserver-xephyr libxml2-dev libxslt-dev libssl-dev z
 # Install OWTF using the recommended method (setup.py)
 RUN cd /owtf &&\
-  python setup.py develop
+  python setup.py install
```
you should keep this to `develop` since this is a dev Docker container - any changes you do while developing will automatically reflect otherwise you have to install the package every time to see your changes.

Ashrith-Shetty replied:
I have made the changes. Thank You.

viyatb commented on docker/Dockerfile:
```diff
@@ -23,38 +23,44 @@ RUN apt-get -y install  xvfb xserver-xephyr libxml2-dev libxslt-dev libssl-dev z
 # Expose the required ports for OWTF to run
 EXPOSE 8008 8009 8010
+
+
```
extra line

Ashrith-Shetty replied:
@viyatb  Have removed the extra lines, Thanks

Ashrith-Shetty committed:
Update Dockerfile

Ashrith-Shetty committed:
Update docker-compose.dev.yml

When RUN python setup.py develop command in the Dockerfile it creates /owtf/.egg-info in the current directory(docker image), but when we volume-mount our host machine's project over the one in the container, it won't have an egg-info file.
So we make Docker to use .egg-info from the container instead of host mounted volume.

Ashrith-Shetty committed:
Update Dockerfile

Changing the kali image to kalilinux/kali, since kalilinux/kali is much more stable.
Removed python-all-dev, python-pip, python-dev since they are no longer needed.

Ashrith-Shetty committed:
Update Dockerfile

Ashrith-Shetty commented:
@viyatb Please let me know if you want any additional changes to be made, or more info about the changes I have made.Thank You.

Ashrith-Shetty committed:
Update Dockerfile

Ashrith-Shetty committed:
Update wait-for-it.sh latest version

Ashrith-Shetty committed:
Added python3-impacker required for plugin PTES-001

viyatb commented on docker/docker-compose.dev.yml:
```diff
@@ -19,6 +19,7 @@ services:
       - db
     volumes:
       - ..:/owtf
+      - /owtf/owtf.egg-info
```
lets remove this - egg info is only builds

Ashrith-Shetty replied:
@viyatb Have made the changes.Thanks

Ashrith-Shetty committed:
Python3.7 is no longer available in Kali Docker Image

Ashrith-Shetty committed:
Update psutil and cffi versions to support python3.8

Ashrith-Shetty committed:
Changed the installation method (setup.py install)

Ashrith-Shetty committed:
Not mounting .egg files

viyatb committed:
Merge branch 'develop' into fix-docker-build

Ashrith-Shetty committed:
Added Amass and Go which is need for Subdomain Takeover Plugin
